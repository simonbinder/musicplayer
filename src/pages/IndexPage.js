import React from 'react';
import ResultEntry from '../components/ResultEntry';

export default class IndexPage extends React.Component {
  constructor(props) {
    super(props);
  };

  render() {
    console.log('Store', this.props.store);

    return <div>

      <div className="container">
        <h1>Search results</h1>

        <div className="row">
          <ResultEntry
            origin="spotify"
            title="New Rules"
            artists="Dua Lipa"
          />
          <ResultEntry
            origin="spotify"
          />
          <ResultEntry
            origin="spotify"
          />
        </div>

      </div>

    </div>
  };
};
