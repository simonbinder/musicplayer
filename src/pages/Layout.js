import React from 'react';
import HeaderBar from '../components/HeaderBar';
import '../assets/Main.scss';

export default class Layout extends React.Component {
  constructor(props) {
    super(props);
  };

  onAccountClicked(ev) {
    
  }

  render() {
    return <div>
      <HeaderBar />

      <div className="main">
        {this.props.children}
      </div>

      <div className="footer">
        Footer
      </div>
    </div>
  };
};
