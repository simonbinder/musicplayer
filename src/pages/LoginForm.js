import React from 'react';
import InputField from '../components/InputField';
import InputFieldError from '../components/InputFieldError';
import { loginRequest } from '../services/accountService';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { savePlaylistsInitial } from '../actions/playlistActions';
import '../assets/LoginForm.scss';

export class LoginForm extends React.Component {
  constructor(props) {
      super(props);
      this.state = {
        email: '',
        emailError: '',
        password: '',
        passwordError: '',
      };
      this.handleSubmit = this.handleSubmit.bind(this);
    };

    checkValid(key, checkValue) {
      switch(key) {
        case 'email': {
          if(checkValue) {
            return '';
          }else {
            return 'Please enter a valid Username';
          }
          break;
        }
        case 'password':
          if(checkValue){
            return '';
          }else{
            return 'Please enter your password';
          }
      }
    }

    handleChange(ev, key) {
      var errorKey = key + 'Error';
      this.setState({
        [key]: ev.currentTarget.value,
      });
      var error = this.checkValid(key, ev.currentTarget.value);

      this.setState({
        [errorKey]:error,
      })
    };

  handleSubmit(ev) {

    const {
      email,
      password
    } = this.state;

    var valid = (email != '' || email != null)
      && (password != '' || password != null);

    if(valid) {
      loginRequest(email, password)
      .then(response => {
        console.log('Login response', response);
        //save token
        localStorage.setItem('token', response.token);
        //
        this.props.savePlaylistsInitial(response.playlists);
        //go to indexpage
        this.props.router.push('/');

      })
      .catch(error => {
        console.log('Error', error);
      })
    } else {
      console.log('Form not valid');
    }
  };

  render() {
    return <div className="container">
    <div className="col align-self-center">
      <div className="form-group">
      <label>
        Username:
      </label>
      <InputField
        type="email"
        placeHolder="Bitte den Usernamen eingeben"
        value={this.state.email}
        handleChange={ ev => this.handleChange(ev, 'email') }
      />
      <InputFieldError
        error={this.state.emailError}
      />
      </div>
      <div className="form-group">
        <label>
          Password:
        </label>
        <InputField
          type="password"
          placeHolder="Bitte das Passwort eingeben"
          value={this.state.password}
          handleChange={ ev => this.handleChange(ev, 'password') }
        />
        <InputFieldError
          error={this.state.passwordError}
        />
        </div>
        <div className="form-group">
          <input
            type="submit"
            className="btn btn-primary o-btn-sbt"
            onClick={ this.handleSubmit }
            value="Einloggen" />
          </div>

          <div>
            <Link className="btn btn-primary o-btn-sbt" to="/register">Don't have an account? Register here</Link>
          </div>
    </div>
    </div>
  };
};

function mapStateToProps(state) {
  return {};
};

const mapDispatchToProps = dispatch => ({
  goToIndexPage: () => dispatch(push('/')),
  savePlaylistsInitial: playlists => dispatch(savePlaylistsInitial(playlists)),
});

//TODO: rename export
export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
