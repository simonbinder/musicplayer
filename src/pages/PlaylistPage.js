import React from 'react';
import { connect } from 'react-redux';
import { selectPlaylist } from '../selector/playlistSelector';
import TrackBox from '../components/TrackBox';
import { startNewSong } from '../actions/playlistActions';

class PlaylistPage extends React.Component {
  constructor(props) {
    super(props);
    this.onSongPlay = this.onSongPlay.bind(this);
  };

  onSongPlay(track) {
    this.props.onSongPlay(track);
  };

  render() {

    console.log('Playlist', this.props.playlist);
    console.log('ownProps', this.props.ownProps);

    if(!this.props.playlist) {
      return <div className="container">
        <h2>Playlist does not exist</h2>
      </div>
    }

    return <div className="container">
      <div className="row">
        { this.props.playlist.tracks.map((track, key) => {
          return <TrackBox
            title={track.title}
            artists={track.artists}
            key={key}
            onPlay={ () => this.onSongPlay(track) }
          />
        })}
      </div>
    </div>
  };
};

function mapStateToProps(store, ownProps) {
  return {
    //playlist: store.playlist,
    playlist: selectPlaylist(store.playlist.playlists, ownProps.location.query.id),
    //ownProps: ownProps,
  };
};

const mapDispatchToProps = dispatch => ({
  onSongPlay: track => dispatch(startNewSong(track)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PlaylistPage);
