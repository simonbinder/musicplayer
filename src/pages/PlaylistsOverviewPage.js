import React from 'react';
import {Â connect } from 'react-redux';
import PlaylistBox from '../components/PlaylistBox';
import { createNewPlaylist } from '../actions/playlistActions';
import { Link } from 'react-router';

class PlayListsOverviewPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      playlistName: '',
    };
    this.onPlaylistNameChange = this.onPlaylistNameChange.bind(this);
    this.onPlaylistCreateSubmit = this.onPlaylistCreateSubmit.bind(this);
  };

  onPlaylistNameChange(ev) {
    this.setState({
      playlistName: ev.currentTarget.value,
    });
  };

  onPlaylistCreateSubmit(ev) {
    const {
      user
    } = this.props.credentials;

    this.props.onPlaylistCreateSubmit(user._id, this.state.playlistName);
  }

  render() {
    return <div className="container">
      <h2>Create a new playlist:</h2>
      <input type="text" value={this.state.playlistName} onChange={ ev => this.onPlaylistNameChange(ev) } />
      <button onClick={ this.onPlaylistCreateSubmit }>Create</button>

      <h2>Your playlists</h2>

      { this.props.playlist.playlists.map((playlist, key) => {
        return <Link key={key} to={ '/playlist?id=' + playlist._id }>
        <PlaylistBox
          key={key}
          name={playlist.name}
        />
        </Link>
      })}
    </div>
  };
};

function mapStateToProps(store) {
  return {
    playlist: store.playlist,
    credentials: store.credentials,
  };
};

const mapDispatchToProps = dispatch => ({
  onPlaylistCreateSubmit: (id, name) => dispatch(createNewPlaylist(id, name)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PlayListsOverviewPage);
