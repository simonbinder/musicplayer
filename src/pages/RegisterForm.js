import React from 'react';
import '../assets/RegisterForm.scss';
import InputFieldError from '../components/InputFieldError';
import InputField from '../components/InputField';

export default class RegisterForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      emailError: '',
      password: '',
      passwordError: '',
      confirmPassword: '',
      confirmPasswordError: '',
    };
  };

  checkValid(key, checkValue) {
    switch(key) {
      case 'email': {
        if(checkValue && checkValue != '' && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(checkValue)) {
          return 'Email is not valid';
        } else {
          return '';
        }
        break;
      }
      case 'password': {
        if(checkValue && /^(?=.*\d).{4,8}/.test(checkValue)) {
          return '';
        } else {
          return 'Password must be between 4 and 8 digits long and include at least one numeric digit.';
        }
        break;
      }
      case 'confirmPassword': {
        if(checkValue == this.state.password && checkValue.length > 0) {
          return '';
        } else {
          return "Passwords don't match";
        }
        break;
      }
    }
  }

  handleChange(ev, key) {
    var errorKey = key + 'Error';
    this.setState({
      [key]: ev.currentTarget.value,
    });
    var error = this.checkValid(key, ev.currentTarget.value);

    this.setState({
      [errorKey]: error,
    })
  };

  handleSubmit(ev) {
    const {
      email,
      password
    } = this.state;

  };

  render() {
    return <div className="c-register-container">
      <div className="form-group">
        <label>
          Username:
        </label>
        <InputField
          type="email"
          placeHolder="Bitte den Usernamen eingeben"
          value={this.state.email}
          handleChange={ ev => this.handleChange(ev, 'email') }
        />
        <InputFieldError
          error={this.state.emailError}
        />
    </div>
    <div className="form-group">
      <label>
        Passwort:
      </label>
      <InputField
        type="password"
        placeHolder="Bitte das Passwort eingeben"
        value={this.state.password}
        handleChange={ ev => this.handleChange(ev, 'password') }
        />
      <InputFieldError
        error={this.state.passwordError}
      />
    </div>
    <div className="form-group">
      <label>
        Passwort wiederholen:
      </label>
      <InputField
      type="password"
      placeHolder="Bitte das Passwort bestÃ¤tigen"
      value={this.state.confirmPassword}
      handleChange={ ev => this.handleChange(ev, 'confirmPassword') }
      />
      <InputFieldError
        error={this.state.confirmPasswordError}
      />
    </div>
    <div className="form-group">
      <input
        type="submit"
        className="btn btn-primary"
        onClick={ this.handleSubmit }
        value="Submit" />
      </div>
    </div>
  };
};
