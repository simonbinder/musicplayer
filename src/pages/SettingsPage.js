import React from 'react';
import '../assets/Settings.scss';
import {
  saveSpotifyAccessToken
} from '../actions/credentialsActions';
import { connect } from 'react-redux';

class SettingsPage extends React.Component {
  constructor(props) {
    super(props);
  };

  componentDidMount() {
    console.log('Settingspage mount');

    const {
      api
    } = this.props.location.query;

    switch(api) {
      case "spotify":
        console.log('Spotify');
        const {
          access_token,
          refresh_token
        } = this.props.location.query;

        this.props.saveSpotifyAccessToken(access_token);

        console.log('Store refresh_token');
        localStorage.setItem('spotifyRefreshToken', refresh_token);
        break;
    }
  }

  render() {
    console.log('Query ', this.props.location.query);
    console.log('Spotify connected', this.props.credentials.spotifyConnected);

    return <div>
      <div className="container">
        <h1 className="title">Activate and deactive our supported APIs</h1>
        <div className="row">

          <div className="col-md-12 o-settings-block">
            <p>Connect to your soundcloud account:</p>
            <img src="assets/images/soundcloud-connect.png" />
          </div>

          <div className="col-md-12 o-settings-block">
            <div className="o-settings-block__description">
              <p>Connect to your spotify account:</p>
              <a href="http://localhost:4000/spotify/login">Connect to spotify</a>
            </div>
            <div className="o-settings-block__status">
              <span style={{ color: 'green' }}>Connected</span>
            </div>
          </div>

        </div>
      </div>
    </div>
  };
};

function mapStateToProps(store) {
  return {
    credentials: store.credentials,
  };
};

const mapDispatchToProps = dispatch => ({
  saveSpotifyAccessToken: token => dispatch(saveSpotifyAccessToken(token)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SettingsPage);
