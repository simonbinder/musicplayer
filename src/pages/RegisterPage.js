import React from 'react';
import RegisterForm from './RegisterForm';
import '../assets/RegisterForm.scss';

export default class RegisterPage extends React.Component {
  constructor(props) {
    super(props);
  };

  render() {
    return <div className="container">
        <RegisterForm></RegisterForm>
      </div>
  };
};
