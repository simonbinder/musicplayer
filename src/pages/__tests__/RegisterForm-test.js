import React from 'react';
import {RegisterForm} from '../../pages/RegisterForm';
import InputFieldError from '../../components/InputFieldError';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('RegisterForm Component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<RegisterForm/>).toJSON();
    expect(tree).toMatchSnapshot();
  }),

  test('No error message displayed when no error state set', () => {
    const loginComponent = shallow(<RegisterForm />);
    const inputFieldError = (loginComponent.find(InputFieldError).first()).shallow();
    expect(inputFieldError.find('.o-error-active').length).toBe(0);
  }),

  test('Error message displayed when error state set', () => {
    const loginComponent = shallow(<RegisterForm />);
    loginComponent.setState({ emailError: 'Please enter a valid Username' });
    const inputFieldError = (loginComponent.find(InputFieldError).first()).shallow();
    expect(inputFieldError.find('.o-error-active').length).toBe(1);
  })

  test('No url in email', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('email', 'asdf')).toBe('Email is not valid');
  })

  test('Country domain in email missing', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('email', 'asdf@asd')).toBe('Email is not valid');
  })

  test('Correct email entered', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('email', 'test@test.de')).toBe('');
  })

  test('password too short', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('password', '1as')).toBe('Password must be between 4 and 8 digits long and include at least one numeric digit.');
  })

  test('Digit in password missing', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('password', 'asdfas')).toBe('Password must be between 4 and 8 digits long and include at least one numeric digit.');
  })

  test('Password too long', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('password', 'asdffdsasdfas1')).toBe('Password must be between 4 and 8 digits long and include at least one numeric digit.');
  })

  test('Correct password entered', () => {
    const loginComponent = shallow(<RegisterForm />);
    expect(loginComponent.instance().checkValid('password', '123adf')).toBe('');
  })
});
