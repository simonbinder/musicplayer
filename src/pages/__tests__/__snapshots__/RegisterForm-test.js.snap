// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Component Correct email entered 1`] = `""`;

exports[`RegisterForm Component Correct password entered 1`] = `""`;

exports[`RegisterForm Component Country domain in email missing 1`] = `"Email is not valid"`;

exports[`RegisterForm Component Digit in password missing 1`] = `"Password must be between 4 and 8 digits long and include at least one numeric digit."`;

exports[`RegisterForm Component Error message displayed when error state set 1`] = `1`;

exports[`RegisterForm Component No error message displayed when no error state set 1`] = `0`;

exports[`RegisterForm Component No url in email 1`] = `"Email is not valid"`;

exports[`RegisterForm Component Password too long 1`] = `"Password must be between 4 and 8 digits long and include at least one numeric digit."`;

exports[`RegisterForm Component password too short 1`] = `"Password must be between 4 and 8 digits long and include at least one numeric digit."`;

exports[`RegisterForm Component renders correctly 1`] = `
<div
  className="c-register-container"
>
  <div
    className="form-group"
  >
    <label>
      Username:
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Bitte den Usernamen eingeben"
      type="email"
      value=""
    />
    <div
      className="o-error"
    >
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Passwort:
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Bitte das Passwort eingeben"
      type="password"
      value=""
    />
    <div
      className="o-error"
    >
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Passwort wiederholen:
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Bitte das Passwort bestÃ¤tigen"
      type="password"
      value=""
    />
    <div
      className="o-error"
    >
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <input
      className="btn btn-primary"
      onClick={[Function]}
      type="submit"
      value="Submit"
    />
  </div>
</div>
`;
